dropwizard/service/src/test/java/org/apache/polaris/service/dropwizard/PolarisApplicationIntegrationTest.java

```diff
@@ -722,7 +722,7 @@ public void testRequestBodyTooLarge() {
             .header(REALM_PROPERTY_KEY, realm)
             .post(largeRequest)) {
       assertThat(response)
-          .returns(Response.Status.BAD_REQUEST.getStatusCode(), Response::getStatus)
+          .returns(Response.Status.REQUEST_ENTITY_TOO_LARGE.getStatusCode(), Response::getStatus)
```

Comment:
IMHO we do need to take connection closed into account. With Quarkus for sure, maybe even with Dropwizard. From my recollection and manual testing, both frameworks close the connection on this response error. The problem is that sometimes the client receives the response payload before the connection is closed, sometimes after. Again from my recollection, in Quarkus, 1 execution out of 3 or 4 ends up with connection closed.

Reply:
From [RFC 9110](https://httpwg.org/specs/rfc9110.html#status.413) on status 413:

> The server MAY terminate the request, if the protocol version in use allows it; otherwise, the server MAY close the connection.

Reply:
RFC 9110 defines the protocol-level behaviour, but in Polaris, the REST endpoint produces a response object with a status code in this case. I would not think Quarkus closing the connection prematurely would be correct. I believe Quarkus (or any other Web App framework) ought to send the response object to the client.

@adutra : are you sure the connection resets happened in the "body" test and and not in the "header" test? The headers are handled by the Web Apop framework, so the connection reset decision lies with the framework itself. Still, the error on large payload objects is handled differently.

Reply:
> are you sure the connection resets happened in the "body" test and and not in the "header" test? 

Yes 😄 – those two tests gave me enough headaches that I still remember...

> The headers are handled by the Web Apop framework,

In Quarkus, both cases are handled by Vertx.

Reply:
Here is the code in Quarkus that handles large bodies:

https://github.com/quarkusio/quarkus/blob/0d6a99e1dc7d7394adfde2a836442124848d7e81/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/options/HttpServerCommonHandlers.java#L49-L55

As you can see, the connection is closed when the response is done.

Reply:
> In Quarkus, both cases are handled by Vertx

Ah... I see. The the behaviour you observed makes sense.

Is there a config parameter to max body size in Vertx?

Reply:
I think the correct test would be to check the status code, but avoid checking the response payload on 413... WDYT?

Reply:
I think we can fix the header test the same way in #630

Reply:
And [quarkus.http.limits.max-header-size](https://quarkus.io/guides/http-reference#quarkus-vertx-http_quarkus-http-limits_quarkus-http-limits-max-header-size) for the headers size.

Reply:
No, I do not think so. The HTTP protocol requires the status like to be sent, so on the client side the connection can be closed only after the status like is received. The response body is not guaranteed by the RFC, though.

Reply:
> on the client side the connection can be closed only after the status line is received

Where is that specified? I find it hard to enforce such a guarantee. If the server closes the connection, I don't think there is a way for the TCP layer to close the connection gracefully only after some TCP message containing the status is received by the client.

Reply:
RFC 9110 Section [3.4](https://httpwg.org/specs/rfc9110.html#messages) `A server responds to a client's request by sending one or more response messages, each including a status code [...]`

Reply:
Sure, but from [RFC 9112](https://httpwg.org/specs/rfc9112.html#persistent.tear-down):

> If a server performs an immediate close of a TCP connection, there is a significant risk that the client will not be able to read the last HTTP response. If the server receives additional data from the client on a fully closed connection, such as another request sent by the client before receiving the server's response, the server's TCP stack will send a reset packet to the client; unfortunately, the reset packet might erase the client's unacknowledged input buffers before they can be read and interpreted by the client's HTTP parser.

Reply:
What an interesting :) mess, TIL... In this case, I'd say we should drop both the body and header size tests, because it's way beyond the scope of Polaris to ensure deterministic behaviours of these tests.

I personally do not see much value in a test that _either_ validates the status code or does not validate anything because a connection close exception is inconclusive and does not guarantee that even the request was sent.

Reply:
Yeah... the more it goes the more I think it makes sense to drop both tests.